### PR TITLE
Google demo

### PR DESCRIPTION
**Pull request by nhavronskyi**

### CODE REVIEW

File: uniScheduleBot/src/main/java/com/example/unischedulebot/component/Bot.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import java.util.stream.Collectors;
+
+@Component
+@RequiredArgsConstructor
+public class Bot extends TelegramLongPollingBot {
+
+    private final TelegramProps telegramProps;
+
+    private final GService gService;
+    private Long chatId;
```
Comment (syevtushok) [inline uniScheduleBot/src/main/java/com/example/unischedulebot/component/Bot.java]: do you need this variable?

File: uniScheduleBot/src/main/java/com/example/unischedulebot/component/Bot.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                    "getEvents", "getEvents",
+                    "createDemoEvent", "createDemoEvent",
+                    "creatEventsForSchedule", "creatEventsForSchedule"
+            ), update.getMessage().getChatId());
+            sendApiMethodAsync(sendMessage);
+            chatId = update.getMessage().getChatId();
+        }
+        if (update.hasCallbackQuery()) {
+            if (update.getCallbackQuery().getData().equals("getSchedule")) {
+                String schedule = gService.getSchedule().entrySet()
```
Comment (syevtushok) [inline uniScheduleBot/src/main/java/com/example/unischedulebot/component/Bot.java]: move this logic to the separate service

File: uniScheduleBot/src/main/java/com/example/unischedulebot/component/Bot.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            }
+            if (update.getCallbackQuery().getData().equals("creatEventsForSchedule")) {
+                gService.creatEventsForSchedule();
+                sendApiMethodAsync(sendMessage("events from the schedule have been created", chatId));
+            }
+        }
+    }
+
+    private SendMessage sendMessage(String message, Long chatId) {
+        SendMessage sendMessage = new SendMessage();
```
Comment (syevtushok) [inline uniScheduleBot/src/main/java/com/example/unischedulebot/component/Bot.java]: move this logic to the separate service

File: uniScheduleBot/src/main/java/com/example/unischedulebot/google/GCalendar.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import com.google.api.services.calendar.model.Events;
+import lombok.SneakyThrows;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@Service
+public class GCalendar {
```
Comment (syevtushok) [inline uniScheduleBot/src/main/java/com/example/unischedulebot/google/GCalendar.java]: create an interface for the service

File: uniScheduleBot/src/main/java/com/example/unischedulebot/google/GSheets.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+
+@Service
+public class GSheets {
+
+    public LinkedHashMap<String, List<String>> getSchedule() {
+        LinkedHashMap<String, List<String>> linkedHashMap = new LinkedHashMap<>();
```
Comment (syevtushok) [inline uniScheduleBot/src/main/java/com/example/unischedulebot/google/GSheets.java]: use interface

File: uniScheduleBot/src/main/java/com/example/unischedulebot/google/GSheets.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        linkedHashMap.put("MONDAY", dayManager("A3:D8"));
+        linkedHashMap.put("TUESDAY", dayManager("A10:D11"));
+        linkedHashMap.put("WEDNESDAY", dayManager("A13:D14"));
+        linkedHashMap.put("FRIDAY", dayManager("A16:D18"));
+        return linkedHashMap;
+    }
+
+    @SneakyThrows
+    private List<String> dayManager(String range) {
```
Comment (syevtushok) [inline uniScheduleBot/src/main/java/com/example/unischedulebot/google/GSheets.java]: replace @SneakyThrows with your exception

File: uniScheduleBot/src/main/java/com/example/unischedulebot/google/GSheets.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        List<String> SCOPES = Collections.singletonList(SheetsScopes.SPREADSHEETS_READONLY);
+        final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+        final String spreadsheetId = "1iaOhFr9FQP_NGF4y2lnrjj3OAbAm7dyMYWl7pb4d4fs";
```
Comment (syevtushok) [inline uniScheduleBot/src/main/java/com/example/unischedulebot/google/GSheets.java]: hide these properties

File: .idea/GitLink.xml [hunk 8]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="uk.co.ben_gibson.git.link.SettingsState">
```
Comment (syevtushok) [inline .idea/GitLink.xml:3]: Don't commit .idea folder

**Review by syevtushok — CHANGES_REQUESTED** [review]

File: uniScheduleBot/src/main/java/com/example/unischedulebot/component/Bot.java (hunk 2 above)
Comment (nhavronskyi) [reply uniScheduleBot/src/main/java/com/example/unischedulebot/component/Bot.java]: done

File: uniScheduleBot/src/main/java/com/example/unischedulebot/component/Bot.java (hunk 3 above)
Comment (nhavronskyi) [reply uniScheduleBot/src/main/java/com/example/unischedulebot/component/Bot.java]: done

File: uniScheduleBot/src/main/java/com/example/unischedulebot/google/GCalendar.java (hunk 4 above)
Comment (nhavronskyi) [reply uniScheduleBot/src/main/java/com/example/unischedulebot/google/GCalendar.java]: done

File: uniScheduleBot/src/main/java/com/example/unischedulebot/google/GSheets.java (hunk 5 above)
Comment (nhavronskyi) [reply uniScheduleBot/src/main/java/com/example/unischedulebot/google/GSheets.java]: done

File: uniScheduleBot/src/main/java/com/example/unischedulebot/google/GSheets.java (hunk 7 above)
Comment (nhavronskyi) [reply uniScheduleBot/src/main/java/com/example/unischedulebot/google/GSheets.java]: done

File: .idea/GitLink.xml (hunk 8 above)
Comment (nhavronskyi) [reply .idea/GitLink.xml:3]: done

**Pull request closed without merging**